miranda/units.py

```diff
@@ -79,19 +80,29 @@ def _transform(s):
 
 def get_time_frequency(d: xr.Dataset):
     freq = xr.infer_freq(d.time)
-    if freq is None:
-        raise TypeError()
+
+    # Hacky workaround for irregular Monthly data
+    if freq is None or (1 < int(freq[:-1]) < 32 and freq.endswith("D")):
+        if (
+            (d.diff("time") < pd.Timedelta(32, "D"))
+            & (d.diff("time") > pd.Timedelta(27, "D"))
+        ).all():
+            freq = "1M"
+        else:
+            raise TypeError()
+
     offset = [int(parse_offset(freq)[0]), parse_offset(freq)[1]]
 
     time_units = {
         "s": "second",
         "m": "minute",
         "h": "hour",
         "D": "day",
+        "M": "month",
         "W": "week",
         "Y": "year",
     }
-    if offset[1] in ["S", "M", "H"]:
+    if offset[1] in ["S", "H"]:
```

Comment:
A bit out of this PR's scope, but for `infer_freq` :  "minute" is "T", 
And it will always return "A" for yearly, even in the "YS" case ("AS-JAN").